src/go/gcsrunner/transform_config.go

```diff
@@ -101,29 +98,59 @@ func transformConfigBytes(config []byte, opts FetchConfigOptions) ([]byte, error
 
 func transformEnvoyConfig(bootstrap *bootstrappb.Bootstrap, opts FetchConfigOptions) error {
 	listeners := bootstrap.GetStaticResources().GetListeners()
-	if len(listeners) != 1 {
-		return fmt.Errorf("expected exactly 1 listener, got: %d", len(listeners))
+	if len(listeners) == 0 {
+		return fmt.Errorf("expected at least 1 listener, got: 0")
+	}
+	// Require http_listener or https_listener, as the absence of these incidates
+	// a broken config.
+	// Note that loopback_listener is not required.
+	httpListenerTransformed := false
+	for _, l := range listeners {
+		switch l.GetName() {
+		// The default listener created by this project's configgenerator package.
+		case "http_listener", "https_listener":
+			httpListenerTransformed = true
+			if err := transformHTTPListener(l, opts); err != nil {
+				return fmt.Errorf("failed to transform HTTP listener: %v", err)
+			}
+		// Loopback listener can be optionally used to point async requests such as
+		// those sent by Service Control and JWT Authn filters in order to utilize
+		// Envoy's built-in Access Logging features.
+		// This requires the use of another unused port.
+		case "loopback_listener":
+			if err := transformLoopbackListener(l, opts); err != nil {
+				return fmt.Errorf("failed to transform Loopback Listener")
+			}
+		}
 	}
-	if err := doListenerTransform(listeners[0], opts); err != nil {
+
+	if !httpListenerTransformed {
```

Comment:
Should this also be checked for loopback listener? Sounds like you will always need this in your deployment

Reply:
In the future possibly, but at the moment this will cause all configs without the new loopback listener to break. This will result in forward- and backward-incompatible changes, and it's virtually impossible to lock the versions together.

The migration plan here is to enable this runner to use both the old and the new configurations before updating the management plane to incorporate the loopback listener feature.